src/common/arrow/arrow_array_scan.cpp

```diff
@@ -0,0 +1,580 @@
+        arrayBuffer + srcOffset * sizeof(T), cnt * sizeof(T));
+    for (int64_t i = dstOffset; i < dstOffset + cnt; i++) {
+        outputVector->getValue<T>(i) /= scaleFactor;
+    }
+}
+
+void scanArrowArrayDurationScaledUp(const ArrowArray* array, ValueVector* outputVector,
+    ArrowNullMaskTree* mask, int64_t scaleFactor, int64_t srcOffset, int64_t dstOffset,
+    int64_t cnt) {
+    const int64_t* arrayBuffer = ((const int64_t*)array->buffers[1]) + srcOffset;
+    mask->copyToValueVector(outputVector, dstOffset, cnt);
+    for (int64_t i = 0; i < cnt; i++) {
+        if (!mask->isNull(i)) {
+            int64_t curValue = arrayBuffer[i];
+            outputVector->setValue<interval_t>(
+                i + dstOffset, interval_t(0, 0, curValue * scaleFactor));
+        }
+    }
+}
+
+void scanArrowArrayDurationScaledDown(const ArrowArray* array, ValueVector* outputVector,
+    ArrowNullMaskTree* mask, int64_t scaleFactor, int64_t srcOffset, int64_t dstOffset,
+    int64_t cnt) {
+    const int64_t* arrayBuffer = ((const int64_t*)array->buffers[1]) + srcOffset;
+    mask->copyToValueVector(outputVector, dstOffset, cnt);
+    for (int64_t i = 0; i < cnt; i++) {
+        if (!mask->isNull(i)) {
+            int64_t curValue = arrayBuffer[i];
+            outputVector->setValue<interval_t>(
+                i + dstOffset, interval_t(0, 0, curValue / scaleFactor));
+        }
+    }
+}
+
+void scanArrowArrayMonthInterval(const ArrowArray* array, ValueVector* outputVector,
+    ArrowNullMaskTree* mask, int64_t srcOffset, int64_t dstOffset, int64_t cnt) {
+    const int32_t* arrayBuffer = ((const int32_t*)array->buffers[1]) + srcOffset;
+    mask->copyToValueVector(outputVector, dstOffset, cnt);
+    for (int64_t i = 0; i < cnt; i++) {
+        if (!mask->isNull(i)) {
+            int32_t curValue = arrayBuffer[i];
+            outputVector->setValue<interval_t>(i + dstOffset, interval_t(curValue, 0, 0));
+        }
+    }
+}
+
+void scanArrowArrayDayTimeInterval(const ArrowArray* array, ValueVector* outputVector,
+    ArrowNullMaskTree* mask, int64_t srcOffset, int64_t dstOffset, int64_t cnt) {
+    const int64_t* arrayBuffer = ((const int64_t*)array->buffers[1]) + srcOffset;
+    mask->copyToValueVector(outputVector, dstOffset, cnt);
+    for (int64_t i = 0; i < cnt; i++) {
+        if (!mask->isNull(i)) {
+            int64_t curValue = arrayBuffer[i];
+            int32_t day = curValue;
+            int64_t micros = (curValue >> (4 * sizeof(int64_t))) * 1000;
+            // arrow stores ms, while we store us
+            outputVector->setValue<interval_t>(i + dstOffset, interval_t(0, day, micros));
+        }
+    }
+}
+
+void scanArrowArrayMonthDayNanoInterval(const ArrowArray* array, ValueVector* outputVector,
+    ArrowNullMaskTree* mask, int64_t srcOffset, int64_t dstOffset, int64_t cnt) {
+    const int64_t* arrayBuffer =
+        ((const int64_t*)array->buffers[1]) + srcOffset * 2; // technically 16 bits per value
```

Comment:
hmm this is a bit confusing here. is the comment `technically 16 bits per value` on the constant value `2`? the code doesn't seem to be shifting 2 bytes per values, unless I missed something here. do you mean `((const int64_t*)(array->buffers[1]) + srcOffset * 2)` or should the arrayBuffer be `int16_t*`?

Reply:
I could have written it more clearly on my end. The pointer is cast to int64_t, so `+ srcOffset * 2` would be equivalent to shifting by `srcOffset * 2 * 8` bytes. I'll write it more clearly.

I also need to cover this with tests. Pandas only started supporting this datatype not too long ago.